lib/internal/source_map/source_map_cache.js

```diff
@@ -73,6 +76,15 @@ function dataFromUrl(basePath, sourceMappingURL) {
   }
 }
 
+// Cache the length of each line in the file that a source map was extracted
+// from. This allows translation from byte offset V8 coverage reports,
+// to line/column offset Source Map V3.
+function lineLengths(content) {
```

Comment:
Would be interesting to benchmark this.

Reply:
To me, this seemed like the best trade off, of space vs., time. An alternative would be to cache the whole source somewhere ... but this grew the cache, on my test, case from 3M to 19M.

Running nodes' own test suite with coverage enabled remains fast. I think once we start testing this on some large TypeScript repos, however, we might find we loop back around and try to find some optimizations.

Out of curiosity, I'll share some numbers from my test cases.

Reply:
I did some testing against the TSLint codebase, which happened to have quite a few tests, and use TypeScript/source-maps:

```
## Baseline

Time (mean ± σ):     19.222 s ±  0.295 s    [User: 31.718 s, System: 1.898 s]
Range (min … max):   18.607 s … 19.744 s    20 runs

## Coverage

Time (mean ± σ):     34.268 s ±  0.552 s    [User: 49.408 s, System: 2.097 s]
Range (min … max):   33.180 s … 35.455 s    20 runs


## Coverage + SourceMap

Time (mean ± σ):     33.862 s ±  0.683 s    [User: 48.863 s, System: 2.099 s]
Range (min … max):   32.488 s … 34.884 s    20 runs

## Coverage + SourceMap + lineLengths

Time (mean ± σ):     34.200 s ±  0.699 s    [User: 49.137 s, System: 2.115 s]
Range (min … max):   32.468 s … 35.811 s    20 runs
```

:point_up_2: it doesn't seem like `lineLengths` slow things down by any noticeable amount ... it does however appear that, with the specific workload, V8's coverage is a significant performance hit (this isn't related to this PR, but is worth having on our minds for future work on this feature).